extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/test/java/io/quarkus/resteasy/reactive/links/deployment/RestLinksInjectionTest.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.resteasy.reactive.links.deployment;
+
+import static io.restassured.RestAssured.when;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import javax.ws.rs.core.Link;
+import javax.ws.rs.core.UriBuilder;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.test.common.http.TestHTTPResource;
+
+public class RestLinksInjectionTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(TestRecord.class, TestResource.class));
+
+    @TestHTTPResource("with-links")
+    String testWithLinksUrl;
+
+    @TestHTTPResource("without-links")
+    String testWithoutLinksUrl;
+
+    @Test
+    void shouldGetWithInstanceLinks() {
+        List<String> links = when().get(testWithLinksUrl + "/1")
+                .thenReturn()
+                .getHeaders()
+                .getValues("Link");
+        assertThat(links).containsOnly(
+                Link.fromUriBuilder(UriBuilder.fromUri(testWithLinksUrl).path("/1")).rel("getById").build().toString(),
```

Comment:
I thought that because we cannot infer all the possible rel values I would simply default it to the method name and allow it to be overwritten with an annotation (`@RestLink(rel="self")`).